mono/metadata/icall.c

```diff
@@ -8504,6 +8503,14 @@ mono_icall_unlock (void)
 void
 mono_icall_cleanup (void)
 {
+	GHashTableIter iter;
+	gpointer key;
+
+	g_hash_table_iter_init (&iter, jit_icall_hash_name);
+	while (g_hash_table_iter_next (&iter, &key, NULL)) {
```

Comment:
This isn't the right way to handle this g_hash_table_destroy will free the keys if you change the init call.